pkg/api/dtos/alerting.go

```diff
@@ -60,7 +60,16 @@ func NewAlertNotification(notification *models.AlertNotification) *AlertNotifica
 		SendReminder:          notification.SendReminder,
 		DisableResolveMessage: notification.DisableResolveMessage,
 		Settings:              notification.Settings,
+		SecureFields:          map[string]bool{},
 	}
+
+	if notification.SecureSettings != nil {
+		for k := range notification.SecureSettings.Decrypt() {
```

Comment:
On second thought, we don't have to decrypt here since we're only using the keys.

```suggestion
		for k := range notification.SecureSettings {
```